lib/composables/Url.ts

```diff
@@ -0,0 +1,98 @@
+import isPlainObject from 'lodash-es/isPlainObject'
+import { watch } from 'vue'
+import { useRoute, useRouter } from 'vue-router'
+
+export interface UseUrlQuerySyncOptions {
+  state: Record<string, any>
+  casts?: Record<string, (value: any) => any>
```

Comment:
Maybe the value could string? Since this value only comes from url query 👀 (doesn't really matter I guess but...

```suggestion
  casts?: Record<string, (value: string) => any>
```